suspect/io/twix.py

```diff
@@ -90,26 +90,55 @@ def calculate_orientation(normal):
     return "SAG"
 
 
-def read_double(name, header_string):
-    substring = re.search(r"<ParamDouble.\"{}\">  {{ <Precision> \d+(  -?[0-9\.]+)?  }}".format(name), header_string)
-    if not substring:
-        raise KeyError(r'ParamDouble."{}" not found in header string'.format(name))
-    number_string = substring.group(1)
-    return float(number_string) if number_string else 0
+def get_meta_regex(regex_list, header_string, convert=1, default=None):
+    """
+    Extract metadata using list of regex and apply unit convertion if value is number.
+    If regex found, but value is empty (usually empty string), will return default.
+
+    Parameters
+    ----------
+    regex_list : List of regex string
+    header_string : TWIX header string
+    convert : Unit convertion, if value is number. Defaults to 1 (means no convertion)
+    default : Default value if match found, but value is empty. Defaults to None.
+
+    Returns
+    -------
+    If number, converted to float and converted to unit (by multiplying by convert).
+    If non-number, return as is. If empty, return default.
+
+    Raises
+    ------
+    KeyError: if no matches found.
+    """
+    rgx_found = False
+    for rgx in regex_list:
+        match = re.findall(rgx, header_string)
+        if len(match) > 0:
+            rgx_found = True
+            match_string = match[-1]
```

Comment:
I just found that there are 2 frequency in Yaps header, the first one sometimes is not the same with frequency in the other header (i.e. `<Param*Frequency`). So I'm taking the last one as it's known to be consistent with the rest. 

Your test data `twix_vb.dat` is a good example of this, there are these 2 frequencies in Yaps header, which is different.
```
sTXSPEC.asNucleusInfo[0].lFrequency      = 123261724
sTXSPEC.asNucleusInfo[0].lFrequency      = 123261716 # consistent with other
```

Reply:
Just a note on header section names - the first one is from MeasYaps header and the second (that is consistent) is from Phoenix header.

Reply:
I opened an issue on our Julia package on this, which contains more detail on this issue (https://github.com/TRIImaging/MagneticResonanceSignals.jl/issues/4), if you'd like to see.